docs/CMakeLists.txt

```diff
@@ -50,20 +48,17 @@ set(DOXYGEN_TOC_INCLUDE_HEADINGS 4)
 set(DOXYGEN_TYPEDEF_HIDES_STRUCT YES)
 set(DOXYGEN_UML_LIMIT_NUM_FIELDS 20)
 set(DOXYGEN_UML_LOOK YES)
-set(DOXYGEN_WARN_NO_PARAMDOC YES)
-set(DOXYGEN_WARN_IF_UNDOCUMENTED YES)
+set(DOXYGEN_WARN_IF_UNDOCUMENTED NO)
 set(DOXYGEN_WARN_IF_DOC_ERROR YES)
-set(DOXYGEN_WARN_AS_ERROR NO)
+set(DOXYGEN_WARN_IF_INCOMPLETE_DOC YES)
+set(DOXYGEN_WARN_NO_PARAMDOC YES)
+set(DOXYGEN_WARN_IF_UNDOC_ENUM_VAL YES)
+set(DOXYGEN_WARN_AS_ERROR FAIL_ON_WARNINGS)
```

Comment:
Are you sure you want this to fail if there are warnings? Won't this prevent you updating the documentation until all the warnings are fixed?

Reply:
But we don't have such warnings for now. Maybe something is wrong in the CI ?

Reply:
There are no warnings raised while generating the docs because `DOXYGEN_EXTRACT_ALL` is set to YES. So actually I suspect that these parameters are meaningless in this file